frontend/src/pages/registration/logInPage.js

```diff
@@ -1,43 +1,102 @@
 
-        fetch('/some-api', { method: form.method, body: formData });
+        const isEmailRegex = /^[^@]$/
+        if (isEmailRegex.test(emailOrUserName))
+            return isValidEmail(emailOrUserName);
 
-        const formJson = Object.fromEntries(formData.entries());
-        console.log(formJson);
+        return isValidUserName(emailOrUserName);
     }
 
-    const [email, setEmail] = useState('');
-    const [password, setPassword] = useState('');
+    async function handleSubmit(e) {
+        e.preventDefault();
+
+        if (!isValidEmailOrUserName(emailOrUserName)) {
+            toast.error("Username or email can not be empty or contain such symbols!");
+            return;
+        }
+
+        if (!isValidPassword(password)) {
+            toast.error("Password must contain at least 8 symbols, where: 1 uppercase letter, 1 lowercase letter and 1 number!");
+            return;
+        }
+
+        try {
+            const response = await axios.post(
+                `${apiEndpoints.url}${apiEndpoints.login.login}`,
+                {
+                    email_or_username: emailOrUserName,
+                    password_hash: password,
+                    auth_provider: globalVariables.authStrategies.simpleStrategy
+                },
+                {
+                    headers: { 'Content-Type': 'application/json' },
+                }
+            );
+
+            login({ email: response?.data?.user?.email, username: response?.data?.user?.username });
+            toast.success(`You are successfully logged in!`);
+            navigate('/');
+        } catch (error) {
+            const errorStatus = error?.response?.status
+            const errorMessage = error?.response?.data?.error;
+            toast.error(
+                `Authentication Error! 
+                ${errorStatus ? errorStatus : ''} 
+                ${errorMessage ? errorMessage : ''}`);
+        }
+    }
 
     return (
         <section className="registration">
             <form method="post" onSubmit={handleSubmit}>
                 <h2>Log In</h2>
                 <p>
-                Email: <input value={email} onChange={e => setEmail(e.target.value)} />
+                    Email/Username:
+                    <input
+                        value={emailOrUserName}
+                        onChange={e => setEmailOrUserName(e.target.value)}
+                        onFocus={ () => toast.info('email: example@email.com') }
+                    />
                 </p>
                 <p>
-                Password: <input type="password" value={password} onChange={e => setPassword(e.target.value)} />
+                    Password:
+                    <input
+                        type="password"
+                        value={password} onChange={e => setPassword(e.target.value)}
+                        onFocus={ () => toast.info('Password must contain at least 8 symbols, where: 1 uppercase letter, 1 lowercase letter and 1 number!') }
```

Comment:
We have repeated messages.

Maybe we can somehow reuse them. Dunno

And the exclamation mark again